api/channel.go

```diff
@@ -0,0 +1,90 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/bwmarrin/discordgo"
+	"github.com/discordomo/api/router/middleware"
+	"github.com/gin-gonic/gin"
+	"github.com/sirupsen/logrus"
+)
+
+// Channel is the struct for channels
+type Channel struct {
+	Name string
+	Type string
+}
+
+// CreateChannel creates a channel
+func CreateChannel(c *gin.Context) {
+	channel := new(Channel)
+	err := c.Bind(channel)
+	// Error message did not return in Insomnia
+	if err != nil {
+		retErr := fmt.Errorf("Unable to parse json body: %w", err)
+
+		c.Error(err)
+		c.AbortWithStatusJSON(http.StatusBadRequest, retErr)
+
+		return
+	}
+
+	dg := middleware.RetrieveSession(c)
+
+	cType := parseChannelType(channel.Type)
+	guildID := os.Getenv("GUILD_ID")
+
+	_, err = dg.GuildChannelCreate(guildID, channel.Name, cType)
+	if err != nil {
+		return
+	}
+
+	resp := fmt.Sprintf("created %s channel %s", channel.Type, channel.Name)
+	c.JSON(http.StatusOK, resp)
+
+	dg.Close()
+}
+
+// DeleteChannel deletes a channel
+func DeleteChannel(c *gin.Context) {
+
+	channel := c.Param("channel")
+
+	dg := middleware.RetrieveSession(c)
+
+	guildID := os.Getenv("GUILD_ID")
```

Comment:
Could we check that `GUILD_ID` is set?

You might want to use lookupenv instead of getenv